src/tiledb/cloud/soma/ingest.py

```diff
@@ -103,87 +102,54 @@ def run_ingest_workflow_udf(
-        )
-
-        collector = grf.submit(
-            lambda output_uri: output_uri,
-            output_uri=output_uri,
-        )
-
-        for entry_input_uri in vfs.ls(input_uri):
-            logger.debug("Processing folder item: entry_input_uri=%r", entry_input_uri)
+        for input_item in vfs.ls(input_uri):
+            logger.debug(
+                "Filtering directory items: input_uri=%r, input_item=%r, pattern=%r",
+                input_uri,
+                input_item,
+                pattern,
+            )
 
             # Subdirectories/subfolders can't be ingested.
-            if vfs.is_dir(entry_input_uri):
-                logger.info(
-                    "Skipping sub-directory: entry_input_uri=%r", entry_input_uri
-                )
-                continue
-
-            base = os.path.basename(entry_input_uri)
-            base, _ = os.path.splitext(base)
-
-            entry_output_uri = output_uri + "/" + base
-            if not output_uri.endswith("/"):
-                entry_output_uri += "/"
-            entry_output_uri += base
-
-            if pattern is not None and not re.search(pattern, entry_input_uri):
-                logger.info(
-                    "Skipping non-matching input: pattern=%r, entry_input_uri=%r",
-                    pattern,
-                    entry_input_uri,
-                )
-                continue
-
-            logger.debug("Submitting h5ad file: entry_input_uri=%r", entry_input_uri)
-
-            node = grf.submit(
-                ingest_h5ad,
-                output_uri=entry_output_uri,
-                input_uri=entry_input_uri,
-                measurement_name=measurement_name,
-                extra_tiledb_config=extra_tiledb_config,
-                ingest_mode=ingest_mode,
-                platform_config=platform_config,
-                resources=ingest_resources,  # Apply propagated resources here.
-                access_credentials_name=carry_along.get("access_credentials_name", acn),
-                logging_level=logging_level,
-                dry_run=dry_run,
-            )
-            collector.depends_on(node)
+            # Use the pattern ".h5ad" to select only .h5ad files.
+            if not vfs.is_dir(input_item) and (
+                not pattern or re.search(pattern, input_item)
+            ):
+                logger.debug("Identified input file: input_item=%r", input_item)
+                input_files.append(input_item)
 
     elif vfs.is_file(input_uri):
-        name = ("dry-run" if dry_run else "ingest") + "-h5ad-file"
+        input_files.append(input_item)
+    else:
+        raise ValueError("input_uri %r is neither a file nor a directory", input_uri)
 
-        grf = dag.DAG(
-            name=name,
-            mode=dag.Mode.BATCH,
-            namespace=carry_along.get("namespace", namespace),
-        )
+    logger.info("Building DAG for SOMA ingestion: input_files=%r", input_files)
```

Comment:
After we've made the input files list, we only need one branch of code to set up and execute this UDF's task graph.